cardano-testnet/src/Testnet/Components/Configuration.hs

```diff
@@ -78,35 +79,32 @@ createConfigYaml (TmpAbsolutePath tempAbsPath') anyCardanoEra' = GHC.withFrozenC
 numSeededUTxOKeys :: Int
 numSeededUTxOKeys = 3
 
+newtype NumPools = NumPools { unNumPools :: Int }
+
 createSPOGenesisAndFiles
   :: (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
-  => CardanoTestnetOptions
-  -> UTCTime -- ^ Start time
+  => NumPools
+  -> AnyCardanoEra -- ^ The era to use
+  -> ShelleyGenesis StandardCrypto -- ^ The shelley genesis to use.
```

Comment:
I did this change to avoid mixing `CardanoTestnetOptions` and genesis files in the same signature. We should avoid having them together in a signature, because they contain duplicate data, e,g, the network magic is both in `CardanoTestnetOptions` and the genesis files.

So if we have them together, we are imposing on the caller to be careful to pass coherent data. This change avoids that, by using smaller types instead.

Reply:
What about the `ConwayGenesis` and `AlonzoGenesis` parameters?

Reply:
We should not expose the data constructor for `NumPools` and expose smart constructors instead: e.g `numPools :: CardanoTestnetOptions -> NumPoolNodes`

Reply:
> What about the ConwayGenesis and AlonzoGenesis parameters?

@Jimbo4350> since `create-testnet-data` doesn't yet take alonzo and conway genesis as CLI parameters, they do not make sense here yet. Adding them is tracked by https://github.com/IntersectMBO/cardano-cli/issues/548

Reply:
> We should not expose the data constructor for NumPools and expose smart constructors instead: e.g numPools :: CardanoTestnetOptions -> NumPoolNodes

Done :+1: